#122 pager widget docs, tests, and some math redoing

JoeZ99 opened the pull request:
@carlosescri this is an itching long due.

I took the liberty of changing the math you used to calculate the page links, please check it out

also removed some sections on the template that didn't make sense.

it's been so long that I no longer remember if I have to rebuild the dist files or not, I hope you can forgive me :-)

carlosescri commented:
@JoeZ99 this will take some time to review, be patient 😄

carlosescri commented:
Your _math_ and template changes break the design.

The idea with the previous and next buttons is to disable them when not used instead of hiding them, so there's as few _blinking_ as possible in the UI. I could accept having an option for the other behavior.

Regarding the _math_, with your approach:

![image](https://user-images.githubusercontent.com/482075/41153490-13da5b0e-6b17-11e8-8da6-8f6964a20277.png)

![image](https://user-images.githubusercontent.com/482075/41153501-1da96a8a-6b17-11e8-88c7-728f06a93716.png)

What's between the `1` and `2` pages? `1.5`? :wink:

My math may not be perfect, but the result makes sense:

![image](https://user-images.githubusercontent.com/482075/41153542-4d7c24dc-6b17-11e8-9378-38577ae1592a.png)

![image](https://user-images.githubusercontent.com/482075/41153554-5796dda4-6b17-11e8-803e-b59e694a87fc.png)

JoeZ99 commented:
@carlosescri

I wasn't aware of the three dots thing: sorry!!

I've reverted templates, but instead if
```html
          {{^previous}}
          <span data-page="{.}">
            {{#translate}}{{previousLabel}}{{/translate}}
          </span>
          {{/previous}}
```

i've put
```html
          {{^previous}}
          <span>
            {{#translate}}{{previousLabel}}{{/translate}}
          </span>
          {{/previous}}
```

because `{{.}}` on the `{{^previous}}` context refers to the containing object (`pager`) in it's reflected on the template as an ugly `Object [Object]`. also there is no css rule that needs `data-page` attribute for anything , just the link capture selector (which doesn't apply here).

Also, I've slighly changed your math, because I don't quite understand this line

```coffees
    if (page - @options.delta) <= @options.delta
```

I've replaced it by:

```coffee
    if (page - @options.delta - 1) <= firstPage
```

which I understand addresses the `...` issue as well. (the other I frankly don't understand its math). but I know some math expressions can be hard to read, so please excuse me and enlighten me if I'm missing someting.
